airflow/triggers/external_task.py

```diff
@@ -0,0 +1,121 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
```

Comment:
Thank you for great implementation os the trigger, i have updated the sensor to use it. 
However, after some testing i have found some moments that could be improved in the trigger:
1. There is no possibility to process several external tasks at once, using external_task_ids parameter. For that reason ExternalTaskSensor now supports only processing one task per run. Can you please take a look on it and add the possibility?;
2. If you pass name of the Dag that does not exist at all, the trigger will run infinitely trying to query task from that Dag. I have added one additional parameter to set a deadline for this search to prevent infinitive loop. If it will not find specific Dag for a minute, it will terminate.